### ISSUE TITLE
Fix S3440 FP: Variable assignment and switch expression

### ISSUE THREAD

**Issue by a10r** [SonarSource/sonar-dotnet#5246]
### Description

We believe we found a false positive of rule S3440 ("Remove this useless conditional"). Consider the following method:

```csharp
public static Uri ToWebsocketUri(this Uri uri)
{
	var builder = new UriBuilder(uri);
	builder.Scheme = uri.Scheme switch
	{
		"https" => "wss",
		"http" => "ws",
		"wss" => "wss", // <--
		"ws" => "ws", // <--
		_ => throw new ArgumentException($"Cannot convert URI scheme '{uri.Scheme}' to a websocket scheme."),
	};
	return builder.Uri;
}
```

S3440 is raised at the two marked lines. We believe S3440 is incorrect here since these two conditionals cannot be removed without changing the behaviour of the code.

After some testing, we think that this FP occurs whenever there is a switch expression with a conditional of the form `X => X` and the result is assigned to any assignment target, unless the assignment target is a newly declared local variable. For example, S3440 is not raised when we change the method to the following:

```csharp
public static Uri ToWebsocketUri2(this Uri uri)
{
	var builder = new UriBuilder(uri);
	string scheme = uri.Scheme switch
	{
		"https" => "wss",
		"http" => "ws",
		"wss" => "wss",
		"ws" => "ws",
		_ => throw new ArgumentException($"Cannot convert URI scheme '{uri.Scheme}' to a websocket scheme."),
	};
	builder.Scheme = scheme;
	return builder.Uri;
}
```

### Related information

C#/VB.NET Plugins version: SonarLint for Visual Studio 2019 (5.3.0.41207)
Visual Studio version: Visual Studio 2019 (16.11.8)
MSBuild / dotnet version: .NET Core 3.1
SonarScanner for .NET version (if used): no
Operating System: Windows 10

**Comment by martin-strecker-sonarsource** [comment]
Another example
```cs
string M(string s) =>
    s switch
    {
        null => null, // S3440 <- FP
        "" => "Empty",
        var other => "Not empty",
    };
```

**Comment by antoinebj** [comment]
Any plans on fixing this?

**Comment by martin-strecker-sonarsource** [comment]
Replaced by (internal ticket) https://sonarsource.atlassian.net/browse/NET-1278

**Comment by mary-georgiou-sonarsource** [comment]
Fixed in sonar-dotnet 10.9
